tools/operator/testclient/operations/instantiatechaincode.go

```diff
@@ -281,80 +327,149 @@ func (i InstantiateCCUIObject) queryInstalledusingCLI(instantiateObject Instanti
 func (i InstantiateCCUIObject) approveCCusingCLI(instantiateObject InstantiateCCUIObject) error {
 
 	var packageID string
-	orgName := instantiateObject.ChannelOpt.OrgName[0]
-	currentDir, err := paths.GetCurrentDir()
-	if err != nil {
-		return err
-	}
-	err = SetEnvForCLI(orgName, instantiateObject.ConnProfilePath, instantiateObject.TLS, currentDir)
-	if err != nil {
-		return err
-	}
-	queryInstalled, _ := i.queryInstalledusingCLI(instantiateObject)
-	for j := 0; j < len(queryInstalled.CC); j++ {
-		if queryInstalled.CC[j].Label == fmt.Sprintf("%s_%s", instantiateObject.ChainCodeID, instantiateObject.ChainCodeVer) {
-			packageID = queryInstalled.CC[j].PackageID
-			break
+	var peerAddress string
+	var ordererAddress string
+	var connProfilePath string
+	for k := 0; k < len(instantiateObject.ChannelOpt.OrgName); k++ {
+		orgName := instantiateObject.ChannelOpt.OrgName[k]
+		currentDir, err := paths.GetCurrentDir()
+		if err != nil {
+			return err
+		}
+		if strings.Contains(instantiateObject.ConnProfilePath, ".yaml") || strings.Contains(instantiateObject.ConnProfilePath, ".json") {
+			connProfilePath = instantiateObject.ConnProfilePath
+		} else {
+			connProfilePath = fmt.Sprintf("%s/connection_profile_%s.yaml", instantiateObject.ConnProfilePath, orgName)
+		}
+		connProfConfig, err := GetConnProfileInformationForOrg(connProfilePath, orgName)
+		if err != nil {
+			return err
+		}
+		ordererName, err := fetchOrdererInformation(currentDir)
+		ordererURL, err := url.Parse(connProfConfig.Orderers[ordererName[0]].URL)
+		if err != nil {
+			logger.ERROR("Failed to get orderer url from connection profile")
+			return err
+		}
+		ordererAddress = ordererURL.Host
```

Comment:
`ordererAddress := ordererURL.Host` after you remove the `var` declaration above